### DISCUSSION TITLE
Inpaint with jewelry png image prompt

### DISCUSSION THREAD

**Discussion by N0VASIX** [lllyasviel/Fooocus#2987] [Q&A] [answered]
Hello,
I have a problem with the inpaint of jewels using a png image (without background).
I have the jewel in the right place but it is generated with a black background. When I use the describe function with my png image, it describes it with a black background.

A few examples and my prompt image:

![image](https://github.com/lllyasviel/Fooocus/assets/89125163/2e6c0059-216d-4b5e-a82d-01bb585b29d6)
![image](https://github.com/lllyasviel/Fooocus/assets/89125163/88afd96c-1142-4445-837d-ffa386930a2c)
![image](https://github.com/lllyasviel/Fooocus/assets/89125163/ebe6a925-bbf5-4434-9d34-0116d917b6cb)
![image](https://github.com/lllyasviel/Fooocus/assets/89125163/a0b07e23-4f08-43a5-8178-09f0da2fa434)

**Comment by abzaloff** [comment]
put the jewel in any editor on a white background and use that image

**Reply by ILikeToasters** [reply]
I am not sure that will be the solution. This was with the transparent background. I think it might be how it is being done. 
![2024-05-23_16-12-25_6693](https://github.com/lllyasviel/Fooocus/assets/4357657/2a17a46d-095e-4b11-8330-87d1e61756c7)

**Reply by N0VASIX** [reply]
It doesn't work for me.
<img width="763" alt="Capture d’écran 2024-05-24 201054" src="https://github.com/lllyasviel/Fooocus/assets/89125163/edd313e3-167c-44b6-853c-13e826edab9b">

**Reply by N0VASIX** [reply]
![image](https://github.com/lllyasviel/Fooocus/assets/89125163/4f04ff24-9f6e-4bba-b2ef-e958f72fa34f)

**Reply by ILikeToasters** [reply]
Going to want to generate the image first without the ring and then use inpainting so you can limit the size of the area. That was what I did since you mentioned inpainting in your question. I just generated the image first and then mask the limited area on the finger.

**Reply by ILikeToasters** [reply]
If you don't figure it out I will try to make a quick video later when I get home. And don't expect it to work on the first try. Could take a few. I did 4 images and two worked.

**Reply by N0VASIX** [reply]
This image was generated and the result you see is one of those I've done with inpainting.
Could this be because I don't have the latest version of Fooocus or when I log in I get the latest version automatically?

**Reply by ILikeToasters** [reply]
You should have the latest version automatically. I am going to try to get a video of doing this in a little bit. Been testing out the new update before it goes live.

**Reply by ILikeToasters** [reply]
I made a video for you and even improved on how I was doing it. One issue is since the ring image is tilted it causes the ring to often tilt on the finger but if you mask in the shape of the heart it should force Fooocus to follow that as a guide. Also I adjust the denoise strength so it maintains some of the original finger shape and also the inpaint respective field so it works on a smaller area. You may need to adjust these depending on the image that you are inpainting. Uploaded to YouTube since file too big for here. Hopefully this helps. https://www.youtube.com/watch?v=SC1a5T8vwlU

**Reply by N0VASIX** [reply]
It works, thank you very much for your help.

**Comment by ILikeToasters** [comment, marked as answer]
Jewelry can be difficult but how are you doing it? I was able to do it with normal inpainting and masking part of the finger. These settings. 
<img width="565" alt="Screenshot 2024-05-23 161624" src="https://github.com/lllyasviel/Fooocus/assets/4357657/51fc70ed-1b7f-4575-abbf-98287cc36408">

**Reply by ILikeToasters** [reply]
This was the result. 
![2024-05-23_16-12-25_6693](https://github.com/lllyasviel/Fooocus/assets/4357657/a7fd686c-3d1e-4153-952b-034401e581d9)

**Reply by ILikeToasters** [reply]
Instructions: https://www.youtube.com/watch?v=SC1a5T8vwlU
